From 4f3cbfe81887dde86fe009e5dcc5fb124ba625f8 Mon Sep 17 00:00:00 2001
From: Matthias Vallentin <vallentin@icir.org>
Date: Fri, 20 Dec 2013 12:17:03 -0800
Subject: [PATCH] Use all symbols of the static libc++abi library.

When linking libc++ with our static version of libc+abi, we would like to
include all symbols that applications may need in the future and not just those
used by libc++.
---
 lib/CMakeLists.txt |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index cec0bee..bc3bd5c 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -36,7 +36,11 @@ if (DEFINED LIBCXX_CXX_ABI_DEPS)
 endif()
 
 # Generate library list.
-set(libraries ${LIBCXX_CXX_ABI_LIBRARIES})
+if (LIBCXX_LIBCXXABI_WHOLE_ARCHIVE)
+  set(libraries -Wl,--whole-archive ${LIBCXX_CXX_ABI_LIBRARIES} -Wl,--no-whole-archive)
+else()
+  set(libraries ${LIBCXX_CXX_ABI_LIBRARIES})
+endif()
 append_if(libraries LIBCXX_HAS_PTHREAD_LIB pthread)
 append_if(libraries LIBCXX_HAS_C_LIB c)
 append_if(libraries LIBCXX_HAS_M_LIB m)
-- 
1.7.6.4

